adapters/tlsio_ssl_esp8266.c

```diff
@@ -0,0 +1,712 @@
+{
+    void* result;
+    if(
+        (name == NULL) || (value == NULL)
+    )
+    {
+        LogError("invalid parameter detected: const char* name=%p, const void* value=%p", name, value);
+        result = NULL;
+    }
+    else
+    {
+        if (strcmp(name, "TrustedCerts") == 0)
+        {
+            if(mallocAndStrcpy_s((char**)&result, value) != 0)
+            {
+                LogError("unable to mallocAndStrcpy_s TrustedCerts value");
+                result = NULL;
+            }
+            else
+            {
+                /*return as is*/
+            }
+        }
+        else if (strcmp(name, "x509certificate") == 0)
+        {
+            if (mallocAndStrcpy_s((char**)&result, value) != 0)
+            {
+                LogError("unable to mallocAndStrcpy_s x509certificate value");
+                result = NULL;
+            }
+            else
+            {
+                /*return as is*/
+            }
+        }
+        else if (strcmp(name, "x509privatekey") == 0)
+        {
+            if (mallocAndStrcpy_s((char**)&result, value) != 0)
+            {
+                LogError("unable to mallocAndStrcpy_s x509privatekey value");
+                result = NULL;
+            }
+            else
+            {
+                /*return as is*/
+            }
+        }
+        else
+        {
+            LogError("not handled option : %s", name);
+            result = NULL;
+        }
+    }
+    return result;
+}
+
+/*this function destroys an option previously created*/
+static void tlsio_openssl_DestroyOption(const char* name, const void* value)
+{
+    /*since all options for this layer are actually string copies, disposing of one is just calling free*/
+    if (
+        (name == NULL) || (value == NULL)
+        )
+    {
+        LogError("invalid parameter detected: const char* name=%p, const void* value=%p", name, value);
+    }
+    else
+    {
+        if (
+            (strcmp(name, "TrustedCerts") == 0) ||
+            (strcmp(name, "x509certificate") == 0) ||
+            (strcmp(name, "x509privatekey") == 0)
+        )
+        {
+            free((void*)value);
+        }
+        else
+        {
+            LogError("not handled option : %s", name);
+        }
+    }
+}
+
+/* Codes_SRS_TLSIO_SSL_ESP8266_99_001: [ The tlsio_openssl_retrieveoptions shall not do anything, and return NULL. ]*/
+static OPTIONHANDLER_HANDLE tlsio_openssl_retrieveoptions(CONCRETE_IO_HANDLE handle)
+{
+    OPTIONHANDLER_HANDLE result = NULL;
+    return result;
+}
+
+static const IO_INTERFACE_DESCRIPTION tlsio_openssl_interface_description =
+{
+    tlsio_openssl_retrieveoptions,
+    tlsio_openssl_create,
+    tlsio_openssl_destroy,
+    tlsio_openssl_open,
+    tlsio_openssl_close,
+    tlsio_openssl_send,
+    tlsio_openssl_dowork,
+    tlsio_openssl_setoption
+};
+
+static LOCK_HANDLE * openssl_locks = NULL;
+
+static struct CRYPTO_dynlock_value* openssl_dynamic_locks_create_cb(const char* file, int line)
```

Comment:
Some compilers consider that parameter are variables that you declare, and you must use it in someway. If you do not use this variables, the compiler will throw a compilation error. You can avoid this problem adding something like:

(void)(file, line);

on the beginning of the function implementation.

You need to fix it here and in all the other functions that you do not use some of the parameters.